### PR TITLE
Add new article template for Bioinformatics

### PR DESCRIPTION
**Pull request by ShixiangWang**
To contribute a new article template to this package, please make sure you have done the following things (note that `journalname_article` below is only an example name):

- [x] Unless you have done it in any other RStudio's projects before, please sign the [individual](https://rstudioblog.files.wordpress.com/2017/05/rstudio_individual_contributor_agreement.pdf) or [corporate](https://rstudioblog.files.wordpress.com/2017/05/rstudio_corporate_contributor_agreement.pdf) contributor agreement for a significant pull request (it is fine not to sign it if a PR is only intended to fix a few typos). You can send the signed copy to <jj@rstudio.com>.

- [x] Add the `journalname_article()` function to `R/article.R` if the output format is simple enough, otherwise create a separate `R/journalname_article.R`.

- [x] Add the Pandoc LaTeX template `inst/rmarkdown/templates/journalname_article/resources/template.tex`.

- [x] Add a skeleton article `inst/rmarkdown/templates/journalname_article/skeleton/skeleton.Rmd`.

- [x] Add a description of the template `inst/rmarkdown/templates/journalname_article/template.yaml`.

- [x] Please include the document class file (`*.cls`) if needed, but please do not include standard LaTeX packages (`*.sty`) that can be downloaded from CTAN. Please keep the number of new files absolutely minimal, and also make examples minimal (e.g., if you need a `.bib` example, try to only leave one or two bibliography entries in it, and don't include one hundred items in it without using all of them).

- [x] Update Rd and namespace (could be done by `devtools::document()`).

- [x] Update NEWS.

- [x] Update README with a link to the newly supported journal.

- [x] Add a test to `tests/testit/test-formats.R`.

- [x] Add your name to the list of authors `Authors@R` in DESCRIPTION. You don't need to bump the package version in DESCRIPTION.

Lastly, please try your best to do only one thing per pull request (e.g., if you want to add two output formats, do them in two separate pull requests), and refrain from making cosmetic changes in the code base: https://yihui.name/en/2018/02/bite-sized-pull-requests/

Thank you!

***

I add this new template based on #83, so two authors are added to DESCRIPTION. Please correct my modification if there are something wrong.

This PR can close #83.

Best,
Shixiang

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -28,7 +28,7 @@ Authors@R: c(
   person("Dominik", "Leutnant", role = c("aut", "cph"), email = "leutnant@fh-muenster.de", comment = c(ORCID = "0000-0003-3293-2315")),
   person(family = "MDPI", role = c("aut", "cph")),
   person(family = "Taylor and Francis", role = c("aut", "cph")),
-  person("Oğuzhan", "Öğreden", role = c("aut"), comment = c(ORCID = "0000-0002-9949-3348")),
+  person("O<U+011F>uzhan", "<U+00D6><U+011F>reden", role = c("aut"), comment = c(ORCID = "0000-0002-9949-3348")),
```
Comment (yihui) [inline DESCRIPTION:31]: Please revert this change.

**Comment by ShixiangWang** [comment]
Thanks for your comments

**Comment by ShixiangWang** [comment]
@yihui please see the updates, the files are provided by the journal at https://academic.oup.com/bioinformatics/pages/submission_online. I try to do my best to remove useless files. You can check them.

**Comment by ShixiangWang** [comment]
Please also remember to close #83 if this PR is accepted :).

**Comment by ShixiangWang** [comment]
@yihui According to the description from https://academic.oup.com/bioinformatics/pages/submission_online, bst file is necessary, so I delete csl file.

image

And the pdf is also necessary, in my test that removing the pdf, an error occurred.

File: inst/rmarkdown/templates/bioinformatics_article/skeleton/skeleton.Rmd [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+supplementary_info: |
+  Supplementary data are available at Bioinformatics Online.
+
+output:
+  rticles::bioinformatics_article:
+    citation_package: natbib
+    
+bibliography: bibliography.bib
+# biblio-style: natbib
+# csl: bioinformatics.csl
```
Comment (yihui) [inline inst/rmarkdown/templates/bioinformatics_article/skeleton/skeleton.Rmd:47]: Given that this file has been removed, perhaps this line could also be removed?

File: inst/rmarkdown/templates/bioinformatics_article/skeleton/skeleton.Rmd [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+supplementary_info: |
+  Supplementary data are available at Bioinformatics Online.
+
+output:
+  rticles::bioinformatics_article:
+    citation_package: natbib
+    
+bibliography: bibliography.bib
+# biblio-style: natbib
```
Comment (yihui) [inline inst/rmarkdown/templates/bioinformatics_article/skeleton/skeleton.Rmd:46]: On line `#65` in `template.tex`, the bibliography style has already been hard-coded, so I don't think this `biblio-style` field is necessary here. It can confuse users because it may give them a wrong impression that the `biblio-style` is customizable.

Comment (ShixiangWang) [reply inst/rmarkdown/templates/bioinformatics_article/skeleton/skeleton.Rmd:46]: @yihui Thanks, I agree with you. Could you directly change it in this PR web editor?

**Comment by ShixiangWang** [comment]
@yihui I have done the the contributor agreement. :)

File: inst/rmarkdown/templates/bioinformatics_article/skeleton/skeleton.Rmd (hunk 3 above)
Comment (yihui) [reply inst/rmarkdown/templates/bioinformatics_article/skeleton/skeleton.Rmd:46]: Yes, I will.

**Comment by yihui** [comment]
Okay. I'll let @cderv tweak and merge the PR. Thanks again!

**Comment by cderv** [comment]
I can confirm that `OUP_First_SBk_Bot_8401-eps-converted-to.pdf` is necessary and in used in `bioinfo.cls`. In fact it is looking for `OUP_First_SBk_Bot_8401.eps` and does not found it. When converting to pdf, the pdf version is found, so everything goes well after (except a warning because the eps file is not found) - seems ok as is.

**Comment by cderv** [comment]
Thanks @ShixiangWang ! 

For next time, can you try to create a branch in your repo before sending a PR ? 
It is good practice and easier to have PR sent from feature branch. Thank you !

**Comment by yihui** [comment]
I just saw a NOTE in `R CMD check`: https://travis-ci.org/github/rstudio/rticles/builds/715211132#L877-L884

```
* checking for portable file names ... NOTE
Found the following non-portable file path:
  rticles/inst/rmarkdown/templates/bioinformatics_article/skeleton/OUP_First_SBk_Bot_8401-eps-converted-to.pdf
Tarballs are only required to store paths of up to 100 bytes and cannot
store those of more than 256 bytes, with restrictions including to 100
bytes for the final component.
See section ‘Package structure’ in the ‘Writing R Extensions’ manual.
```

We'll have to either rename `bioinformatics_article` or `OUP_First_SBk_Bot_8401-eps-converted-to.pdf`. We need to cut at least 8 bytes off from the path:

```r
> nchar('rticles/inst/rmarkdown/templates/bioinformatics_article/skeleton/OUP_First_SBk_Bot_8401-eps-converted-to.pdf')
[1] 108
```

**Comment by cderv** [comment]
The pdf file is the one provided with the template. I don't know if it is important to keep the name or not. 
We could also provide the `eps` file instead `OUP_First_SBk_Bot_8401.eps`. It is 398 kb instead of  10kb in size. This eps file be converted at rendering in the pdf file `OUP_First_SBk_Bot_8401-eps-converted-to.pdf` locally in the user folder. 
If we do that we can reduce by more than 8 characters.

Otherwise, we rename the file or we rename the function to `bioinf_article` 😉

**Comment by yihui** [comment]
Perhaps we just get rid of the `_article` suffix in the folder names (which is exactly 8 characters):

```
acm_article             bioinformatics_article  jss_article             rsos_article
acs_article             biometrics_article      mdpi_article            rss_article
aea_article             copernicus_article      mnras_article           sage_article
agu_article             ctex                    oup_article             sim_article
amq_article             elsevier_article        peerj_article           springer_article
ams_article             frontiers_article       plos_article            tf_article
arxiv_article           ieee_article            pnas_article
asa_article             joss_article            rjournal_article
```

The suffix seems to be redundant anyway.

**Comment by cderv** [comment]
Yes that is another way. We can do that. This folder name is only internal the `_article` does not help particularly.
